src/neovim.rs

```diff
@@ -632,23 +514,23 @@ impl Editor for Neovim {
 
 impl Message for NvimMessage {
     fn read(r: &mut impl BufRead) -> Result<Option<NvimMessage>, RpcError> {
-        let mut deserializer = Deserializer::new(r);
-        Ok(Some(Deserialize::deserialize(&mut deserializer).map_err(
-            |e| match e {
-                rmp_serde::decode::Error::InvalidMarkerRead(_)
-                | rmp_serde::decode::Error::InvalidDataRead(_) => {
-                    RpcError::Read(e.description().into())
-                }
-                _ => RpcError::Deserialize(e.description().into()),
-            },
-        )?))
+        let value = read_value(r).map_err(|e| RpcError::Read(e.description().into()))?;
+        log::debug!("< Nvim: {:?}", value);
+        let inner: NvimMessage =
+            from_value(value).map_err(|e| RpcError::Deserialize(e.description().into()))?;
+        let r = Some(inner);
```

Comment:
We never return `None` here, why do we need to keep that `Option`. How about changing to `Result<NvimMessage, RpcError>`

Reply:
It's the signature from the trait Message, the `Option` is there to represents an empty message, in this case (for nvim), we don't have empty messages, so `None` option is not needed.